planck-c/http.c

```diff
@@ -277,13 +277,13 @@ JSValueRef function_http_request(JSContextRef ctx, JSObjectRef function, JSObjec
         // printf("%d bytes, %x\n", body_state.offset, body_state.data);
         if (body_state.data != NULL) {
             if (binary_response) {
-                JSValueRef* bytes = malloc(sizeof(JSValueRef)*body_state.length);
+                JSValueRef* bytes = malloc(sizeof(JSValueRef)*body_state.offset);
```

Comment:
The defect here is that `length` is the size of the (growing) buffer used to accumulate results. Instead, `offset` is the index to the next byte that would be written into that buffer (and is hence the number of accumulated bytes).